add a method to get mtime of a GzHeader as a datetime

nivkner opened the pull request:
fixes #86

alexcrichton commented on src/gz.rs:
```diff
@@ -862,6 +863,17 @@ impl Header {
     }
+
+    /// Returns the most recent modification time as a datetime if time stamp is available.
+    pub fn datetime(&self) -> Option<time::SystemTime> {
```
Perhaps this could be called something like `mtime_as_datetime` to clarify that this is just reading the `mtime` field of the header?

alexcrichton commented on src/gz.rs:
```diff
@@ -862,6 +863,17 @@ impl Header {
         self.mtime
     }
+
+    /// Returns the most recent modification time as a datetime if time stamp is available.
```
Could you expand the documentation here as well to explain when `None` is returned and to detail how the `mtime` field is interpreted?

nivkner replied:
the interpretation of `mtime` is already in the associated method,  so do I also duplicate the parts about year 2038 and local vs universal,  or just the meaning of it as time since the epoch?

alexcrichton commented:
Looks great to me, thanks @nivkner!

nivkner commented:
I fixed my PR according to @alexcrichton's review. I added the explanation needed to use the method: explain the case where there is `None` and what the time stands for. However I didn't want to copy&paste an entire paragraph on the same page, so I just linked to it (which simply highlights it right now).

Also I changed the name. Would have done it in the first place, but I just wasn't sure about the conventions surrounding alternate representations of struct fields.

alexcrichton commented:
Perfect, thanks @nivkner!
